politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -390,11 +395,246 @@ func (p *piPlugin) voteSummary(token []byte) (*ticketvote.SummaryReply, error) {
 	return &sr, nil
 }
 
+// comments requests all comments on a record from the comments plugin.
+func (p *piPlugin) comments(token []byte) (*comments.GetAllReply, error) {
+	reply, err := p.backend.PluginRead(token, comments.PluginID,
+		comments.CmdGetAll, "")
+	if err != nil {
+		return nil, err
+	}
+	var gar comments.GetAllReply
+	err = json.Unmarshal([]byte(reply), &gar)
+	if err != nil {
+		return nil, err
+	}
+	return &gar, nil
+}
+
+// isAncestorOf returns whether the given parentID is an ancestor of the given
+// childID comment, by traveling the comment tree defined by
+// the ParentID of the child comment.
+func (p *piPlugin) isAncestorOf(ancestorID, childID uint32, cs []comments.Comment) bool {
+	if ancestorID == childID {
+		return true
+	}
+	// Convert comments slice to a map
+	commentsMap := make(map[uint32]comments.Comment, len(cs))
+	for _, c := range cs {
+		commentsMap[c.CommentID] = c
+	}
+	// Travel the comment tree to search for an ancestor
+	for {
```

Comment:
It's typically a bad idea to write for loops without a pre-defined break condition because of the potential for an infinite loop bug to be introduced.  Is it possible to write this as `for current.ParentID != 0 {` instead?